core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1258,7 +1258,7 @@ class ReplicaManager(val config: KafkaConfig,
                     .setOffsetLag(getLogEndOffsetLag(log.topicPartition, log.logEndOffset, log.isFuture))
                     .setIsFutureKey(log.isFuture)
                 }.toList.asJava)
-            }.toList.asJava
+            }.filterNot(_.partitions().isEmpty).toList.asJava
 
             new DescribeLogDirsResponseData.DescribeLogDirsResult().setLogDir(absolutePath)
               .setErrorCode(Errors.NONE.code).setTopics(topicInfos)
```

Comment:
Another question: Should we call `setTopics(topicInfos)` if `topicInfos` is empty?